Objects/listobject.c

```diff
@@ -94,6 +94,16 @@ list_preallocate_exact(PyListObject *self, Py_ssize_t size)
     assert(self->ob_item == NULL);
     assert(size > 0);
 
+    /* Since the Python memory allocator has granularity of 16 bytes,
```

Comment:
alignment is 8bytes on 32bit platform.
https://github.com/python/cpython/blob/b86d783a4eff96306f315acf9a6f1aca85d47fc3/Objects/obmalloc.c#L878-L884

So I think we don't need the `else {` block.